includes/class-wc-form-handler.php

```diff
@@ -803,6 +809,31 @@ public static function add_to_cart_action( $url = false ) {
 			return;
 		}
 
+		$nonce_active = apply_filters( 'woocommerce_add_to_cart_nonce_active', false );
+		if ( $nonce_active ) {
+
+			if ( ! isset( $_REQUEST['woocommerce-cart-nonce'] ) && ! isset( $_REQUEST['_wpnonce'] ) ) {
+				//Include notice to confirm an add to cart event should the nonce be missing (simply forwards the link with nonce appended).
+
+				// To get correct add_to_cart_url, we need variation object.
```

Comment:
This block should perhaps have it's own method - to display/get the confirmation notice and output it.